### PR TITLE
Misc. typos (cont. 2)

### LINKED ISSUE #2228: Typo
https://github.com/PointCloudLibrary/pcl/blob/e84fdcded81a937d31b45ff51a740b10fe169db0/gpu/octree/src/cuda/octree_iterator.hpp#L121

### PR DESCRIPTION
**Pull request by luzpaz**
Found via `codespell -q 3 -I ../pcl-whitelist.txt --skip="./surface/include/pcl/surface/3rdparty,./surface/src/3rdparty,./recognition/include/pcl/recognition/3rdparty"`

Maintainer edit: fixes #2228

### CODE REVIEW

**Comment by SergioRAgostinho** [comment]
There's a merge conflict here. Rebase to the current master to fix it pleas.
```shell 
$ git rebase origin/master
# you're gonna git complaining about conflicts
$ git mergetool
# after you fix the merge conflicts
$ git rebase --continue
```

**Comment by luzpaz** [comment]
done.

File: simulation/src/range_likelihood.cpp [hunk 1]
```diff
@@ -539,7 +539,7 @@ costFunction4(float ref_val,float depth_val)
   float proportion = 0.999f;
   float lhood = proportion + (1-proportion)*(top/bottom);
 
-  // safety fix thats seems to be required due to opengl ayschronizate
+  // safety fix that seems to be required due to opengl asyschronizate
```
Comment (luzpaz) [inline simulation/src/range_likelihood.cpp]: please check these fixes

File: 2d/include/pcl/2d/impl/morphology.hpp [hunk 2]
```diff
@@ -277,7 +277,7 @@ pcl::Morphology<PointT>::subtractionBinary (
     const pcl::PointCloud<PointT> &input1, 
     const pcl::PointCloud<PointT> &input2)
 {
-  const int height = (input1.height < input2.hieght) ? input1.height : input2.height;
+  const int height = (input1.height < input2.height) ? input1.height : input2.height;
```
Comment (luzpaz) [inline 2d/include/pcl/2d/impl/morphology.hpp]: please review

File: 2d/include/pcl/2d/impl/morphology.hpp [hunk 3]
```diff
@@ -299,7 +299,7 @@ pcl::Morphology<PointT>::unionBinary (
...
```
Comment (luzpaz) [inline 2d/include/pcl/2d/impl/morphology.hpp]: please review

File: simulation/src/range_likelihood.cpp (hunk 1 above)
Comment (luzpaz) [reply simulation/src/range_likelihood.cpp]: Darn, i left a typo in asyschronizate. But actually after more research (not sure I why I didn't do this the first time) I think the word should actually be `asynchronization`

Comment (luzpaz) [reply simulation/src/range_likelihood.cpp]: Can someone respond to this?

Comment (taketwo) [reply simulation/src/range_likelihood.cpp]: I'm no native speaker, but [Wiktionary has such a word](https://en.wiktionary.org/wiki/asynchronization).

Comment (taketwo) [reply simulation/src/range_likelihood.cpp]: Do you want to change this? Then we can merge.

Comment (SergioRAgostinho) [reply simulation/src/range_likelihood.cpp]: More importantly, who is Hordur? 😆 My guess would also go with asynchronization.

**Comment by luzpaz** [comment]
83ce99d fixes the discussed revisions  
844babb more misc. comment and doxy typos  
d23c294 remaining source typos I just found  
That should be it.

**Pull request merged**